packages/lib/src/vis/shared/DataToHtml.tsx

```diff
@@ -13,12 +12,11 @@ interface Props<T extends Vector3[]> {
 function DataToHtml<T extends Vector3[]>(props: Props<T>) {
   const { points, children } = props;
 
-  const { dataToHtml } = useVisCanvasContext();
-  const camera = useThree((state) => state.camera);
-
-  const htmlPoints = points.map((pt) => {
-    return dataToHtml(camera, pt);
-  }) as MappedTuple<T, Vector3>;
+  const htmlPoints = useCameraState(
+    (camera, { dataToHtml }) =>
+      points.map((pt) => dataToHtml(camera, pt)) as MappedTuple<T, Vector3>,
+    [points]
+  );
```

Comment:
Hehe same trap as usual... Since it's only used in the persisted data selection story, I completely missed it.